sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentClientImpl.java

```diff
@@ -386,10 +390,16 @@ private void initializeGatewayConfigurationReader() {
         // https://msdata.visualstudio.com/CosmosDB/_workitems/edit/332589
     }
 
-    public void init() {
+    public void init(Function<HttpClient, HttpClient> httpClientInterceptor) {
```

Comment:
I suppose this is for testing correct?

any reason `DirectHttpsClientUnderTest` or something similar cannot be used? we already have similar spy classes.


Please see
`DirectHttpsClientUnderTest`
`HttpClientUnderTestWrapper`
`RxDocumentClientUnderTest`

Reply:
Discussed offline - none of the existing TestClinets allows capturing all Http traffic because the Mockito Spy is established only on HttpClients for certain areas (Gateway, DirectHttps etc.) - none of them covers address resolution - or like this apporach all http requests made by RxDocumentClient